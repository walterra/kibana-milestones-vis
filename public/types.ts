import { Milestones } from 'react-milestones-vis';

type MilestonesProps = Pick<
  React.ComponentProps<typeof Milestones>,
  'distribution' | 'aggregateBy' | 'orientation' | 'useLabels'
>;

export interface MilestonesVisParams extends Required<MilestonesProps> {
  indexPatternId: string;
  categoryField: string;
  labelField: string;
  maxDocuments: number;
  sortField: string;
  sortOrder: 'asc' | 'desc';
}
