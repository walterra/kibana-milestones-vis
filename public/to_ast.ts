/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { buildExpression, buildExpressionFunction } from '../../../src/plugins/expressions/public';
import { Vis } from '../../../src/plugins/visualizations/public';
import { MilestonesExpressionFunctionDefinition, VisParams } from './milestones_fn';

export const toExpressionAst = (vis: Vis<VisParams>) => {
  const milestones = buildExpressionFunction<MilestonesExpressionFunctionDefinition>('milestones', {
    ...vis.params,
    indexPatternId: vis.data.indexPattern!.id!,
  });

  const ast = buildExpression([milestones]);

  return ast.toAst();
};
