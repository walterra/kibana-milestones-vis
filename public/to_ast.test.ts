/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import type { Vis } from '../../../src/plugins/visualizations/public';
import { toExpressionAst } from './to_ast';
import { MilestonesVisParams } from './types';

describe('milestones vis toExpressionAst function', () => {
  let vis: Vis<MilestonesVisParams>;

  beforeEach(() => {
    vis = {
      isHierarchical: () => false,
      type: {},
      params: {},
      data: {
        indexPattern: { id: '123' },
        aggs: {
          getResponseAggs: () => [],
          aggs: [],
        },
      },
    } as any;
  });

  it('should match snapshot without params', () => {
    const actual = toExpressionAst(vis);
    expect(actual).toMatchSnapshot();
  });

  it('should match snapshot params fulfilled', () => {
    // set non-default values
    vis.params = {
      indexPatternId: 'the-index-pattern-id',
      distribution: 'top',
      categoryField: 'the-category-field',
      interval: 'year',
      labelField: 'the-label-field',
      maxDocuments: 100,
      orientation: 'vertical',
      showLabels: false,
      sortField: 'the-sort-field',
      sortOrder: 'desc',
    };
    const actual = toExpressionAst(vis);
    expect(actual).toMatchSnapshot();
  });
});
