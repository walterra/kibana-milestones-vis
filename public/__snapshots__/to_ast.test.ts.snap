// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`milestones vis toExpressionAst function should match snapshot params fulfilled 1`] = `
Object {
  "chain": Array [
    Object {
      "arguments": Object {
        "categoryField": Array [
          "the-category-field",
        ],
        "distribution": Array [
          "top",
        ],
        "indexPatternId": Array [
          "123",
        ],
        "interval": Array [
          "year",
        ],
        "labelField": Array [
          "the-label-field",
        ],
        "maxDocuments": Array [
          100,
        ],
        "orientation": Array [
          "vertical",
        ],
        "showLabels": Array [
          false,
        ],
        "sortField": Array [
          "the-sort-field",
        ],
        "sortOrder": Array [
          "desc",
        ],
      },
      "function": "milestones",
      "type": "function",
    },
  ],
  "type": "expression",
}
`;

exports[`milestones vis toExpressionAst function should match snapshot without params 1`] = `
Object {
  "chain": Array [
    Object {
      "arguments": Object {
        "indexPatternId": Array [
          "123",
        ],
      },
      "function": "milestones",
      "type": "function",
    },
  ],
  "type": "expression",
}
`;
